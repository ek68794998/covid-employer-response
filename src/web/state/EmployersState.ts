import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}