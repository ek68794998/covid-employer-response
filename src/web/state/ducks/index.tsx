import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});