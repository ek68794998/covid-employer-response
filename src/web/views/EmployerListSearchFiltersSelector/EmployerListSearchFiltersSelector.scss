@import "../Colors.scss";

$checkbox-size: 16px;

label.EmployerListSearchFiltersSelector {
	align-items: center;
	display: flex;
	padding: 4px;
	position: relative;

	input {
		appearance: none;
		margin-right: 8px;
		outline: none;
		width: $checkbox-size;

		&:checked + span::before {
			background: $link-normal;
			border-color: $link-normal;
		}
	}

	span::before {
		background: $background;
		border: 2px solid $outline-color;
		border-radius: 2px;
		color: $background;
		content: "\2714\fe0e";
		height: $checkbox-size;
		left: 3px;
		line-height: 110%;
		margin: 2px 8px 0 0;
		position: absolute;
		text-align: center;
		transition: 0.1s all;
		width: $checkbox-size;
	}
}
