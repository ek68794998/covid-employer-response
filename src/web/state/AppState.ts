import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}