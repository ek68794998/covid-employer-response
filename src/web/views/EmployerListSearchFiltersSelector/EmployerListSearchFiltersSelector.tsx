import React, { useState } from "react";
import { RouteProps } from "react-router-dom";

import "./EmployerListSearchFiltersSelector.scss";

interface Props extends RouteProps {
	label: string;

	multiselect: boolean;

	onChange: React.ChangeEventHandler<HTMLInputElement>;
}

const EmployerListSearchFiltersSelector: React.FC<Props> = (props: Props): React.ReactElement => {
	const [ id ] = useState(`EmployerListSearchFiltersSelector-${Math.random().toString(16).slice(-8)}`);

	const { label, multiselect, onChange } = props;

	return (
		<span>
			<input
				id={id}
				onChange={onChange}
				type={multiselect ? "checkbox" : "radio"}
			/>
			<label htmlFor={id}>{label}</label>
		</span>
	);
};

export default EmployerListSearchFiltersSelector;
