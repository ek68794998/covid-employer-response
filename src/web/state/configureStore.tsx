import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;