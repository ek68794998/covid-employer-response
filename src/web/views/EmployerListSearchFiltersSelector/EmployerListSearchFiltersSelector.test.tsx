import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerListSearchFiltersSelector from "./EmployerListSearchFiltersSelector";

describe("<EmployerListSearchFiltersSelector />", () => {
	beforeEach(() => {
		jest.spyOn(global.Math, "random").mockReturnValue(0.123456789);
	});

	afterEach(() => {
		(global.Math.random as any).mockRestore();
	});

	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerListSearchFiltersSelector
							initialValue={true}
							label={"foo"}
							multiselect={false}
							onChange={(): void => { /* Do nothing. */}}
						/>
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
