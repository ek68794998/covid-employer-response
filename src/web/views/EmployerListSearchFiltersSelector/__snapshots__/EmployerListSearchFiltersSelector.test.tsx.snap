// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerListSearchFiltersSelector /> renders as checkboxes 1`] = `
Array [
  <span>
    <input
      defaultChecked={true}
      id="EmployerListSearchFiltersSelector-3739635f"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="EmployerListSearchFiltersSelector-3739635f"
    >
      Foo Check 1
    </label>
  </span>,
  <span>
    <input
      defaultChecked={false}
      id="EmployerListSearchFiltersSelector-3739635f"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="EmployerListSearchFiltersSelector-3739635f"
    >
      Foo Check 2
    </label>
  </span>,
  <span>
    <input
      defaultChecked={true}
      id="EmployerListSearchFiltersSelector-3739635f"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="EmployerListSearchFiltersSelector-3739635f"
    >
      Foo Check 3
    </label>
  </span>,
]
`;

exports[`<EmployerListSearchFiltersSelector /> renders as radio buttons 1`] = `
Array [
  <span>
    <input
      defaultChecked={false}
      id="EmployerListSearchFiltersSelector-3739635f"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor="EmployerListSearchFiltersSelector-3739635f"
    >
      Foo Radio 1
    </label>
  </span>,
  <span>
    <input
      defaultChecked={true}
      id="EmployerListSearchFiltersSelector-3739635f"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor="EmployerListSearchFiltersSelector-3739635f"
    >
      Foo Radio 2
    </label>
  </span>,
  <span>
    <input
      defaultChecked={true}
      id="EmployerListSearchFiltersSelector-3739635f"
      onChange={[Function]}
      type="radio"
    />
    <label
      htmlFor="EmployerListSearchFiltersSelector-3739635f"
    >
      Foo Radio 3
    </label>
  </span>,
]
`;
